esi_leap/api/controllers/v1/utils.py

```diff
@@ -65,8 +67,11 @@ def verify_resource_permission(cdict, offer_dict):
 def get_offer(uuid_or_name, status_filter=None):
     if uuidutils.is_uuid_like(uuid_or_name):
         o = offer.Offer.get(uuid_or_name)
-        if o.status == status_filter:
+        if not status_filter or o.status == status_filter:
             return o
+        else:
+            raise exception.OfferNotFound(
+                offer_uuid=uuid_or_name)
```

Comment:
And tests that cover the various possibilities here (including the exception)?

Reply:
done